src/gui/transferlistwidget.cpp

```diff
@@ -490,58 +490,34 @@ void TransferListWidget::previewSelectedTorrents()
     }
 }
 
-void TransferListWidget::setDlLimitSelectedTorrents()
+void TransferListWidget::setSpeedLimitsSelectedTorrents()
 {
-    QVector<BitTorrent::TorrentHandle *> torrentsList;
-    for (BitTorrent::TorrentHandle *const torrent : asConst(getSelectedTorrents())) {
-        if (torrent->isSeed())
-            continue;
-        torrentsList += torrent;
-    }
+    const QVector<BitTorrent::TorrentHandle *> torrentsList = getSelectedTorrents();
     if (torrentsList.empty()) return;
 
-    int oldLimit = torrentsList.first()->downloadLimit();
-    for (BitTorrent::TorrentHandle *const torrent : asConst(torrentsList)) {
-        if (torrent->downloadLimit() != oldLimit) {
-            oldLimit = -1;
-            break;
-        }
-    }
-
-    bool ok = false;
-    const long newLimit = SpeedLimitDialog::askSpeedLimit(
-                this, &ok, tr("Torrent Download Speed Limiting"), oldLimit
-                , BitTorrent::Session::instance()->globalDownloadSpeedLimit());
-    if (!ok) return;
-
-    for (BitTorrent::TorrentHandle *const torrent : asConst(torrentsList)) {
-        qDebug("Applying download speed limit of %ld Kb/s to torrent %s", (newLimit / 1024l), qUtf8Printable(torrent->hash()));
-        torrent->setDownloadLimit(newLimit);
-    }
-}
-
-void TransferListWidget::setUpLimitSelectedTorrents()
-{
-    QVector<BitTorrent::TorrentHandle *> torrentsList = getSelectedTorrents();
-    if (torrentsList.empty()) return;
+    int oldUploadLimit = torrentsList.first()->uploadLimit();
+    int oldDownloadLimit = torrentsList.first()->downloadLimit();
 
-    int oldLimit = torrentsList.first()->uploadLimit();
-    for (BitTorrent::TorrentHandle *const torrent : asConst(torrentsList)) {
-        if (torrent->uploadLimit() != oldLimit) {
-            oldLimit = -1;
+    for (BitTorrent::TorrentHandle *const torrent : torrentsList) {
+        if (torrent->uploadLimit() != oldUploadLimit)
+            oldUploadLimit = -1;
+        if (torrent->downloadLimit() != oldDownloadLimit)
+            oldDownloadLimit = -1;
+        if ((oldUploadLimit == -1) && (oldDownloadLimit == -1))
```

Comment:
I don't think this is more efficient than using 2 loops.

Reply:
I thought it is, that's why i did this but i don't really know. Will revert.